docs/src/literate-tutorials/linear_elasticity.jl

```diff
@@ -1,3 +1,232 @@
+# spatial dimension.
+dim = 2
+order = 1 # linear interpolation
+ip = Lagrange{RefTriangle, order}()^dim # vector valued interpolation
+qr = QuadratureRule{RefTriangle}(1) # 1 quadrature point
+cellvalues = CellValues(qr, ip);
+
+# ### Degrees of freedom
+# For distributing degrees of freedom, we define a `DofHandler`. The `DofHandler` knows that
+# `u` has two degrees of freedom per node because we vectorized the interpolation above.
+dh = DofHandler(grid)
+add!(dh, :u, ip)
+close!(dh);
+
+# ### Boundary conditions
+# Now, we add boundary conditions. We simply support the bottom and the left side and
+# prescribe a displacement upwards on the top edge.
+# The last argument to `Dirichlet` determines which components of the field should be
+# constrained.
+ch = ConstraintHandler(dh)
+add!(ch, Dirichlet(:u, getfaceset(grid, "bottom"), (x, t) -> 0.0, 2))
+add!(ch, Dirichlet(:u, getfaceset(grid, "left"), (x, t) -> 0.0, 1))
+add!(ch, Dirichlet(:u, getfaceset(grid, "top"), (x, t) -> 0.1, 2))
+close!(ch);
+
+# ### Material behavior
+# Next, we need to define the material behavior. Since we use linear elasticity here,
+# we have a linear problem and only need to assemble the stiffness matrix, but not the 
+# residual vector. Consequently we only need the tangent of the stress $\boldsymbol{\sigma}$
+# with respect to the small strain tensor $\boldsymbol{\varepsilon}$ for our element routine.
+#
+# We also operate on a 2-dimensional problem under plain strain conditions here, keep in
+# mind that the plane stress stiffness tensor is defined differently.
+E = 200e3 # Young's modulus [MPa]
+ν = 0.3 # Poisson's ratio [-]
+#md nothing # hide
+
+λ = E*ν / ((1 + ν) * (1 - 2ν)) # 1st Lamé parameter
+μ = E / (2(1 + ν)) # 2nd Lamé parameter
+I = one(SymmetricTensor{2, dim}) # 2nd order unit tensor
+II = one(SymmetricTensor{4, dim}) # 4th order symmetric unit tensor
+∂σ∂ε = 2μ * II + λ * (I ⊗ I); # elastic stiffness tensor
+
+# ### Element routine
+# The stiffness matrix follows from the weak form such that
+# ```math
+# \left(\underline{\underline{K}}\right)_{ij}
+# =
+# \int_\Omega
+#   \left(
+#       \frac{\partial \boldsymbol{\sigma}}{\partial \boldsymbol{\varepsilon}}
+#       :
+#       \boldsymbol{\nabla}^\mathrm{sym} \boldsymbol{N}_j
+#   \right)
+#   :
+#   \boldsymbol{\nabla} \boldsymbol{N}_i
+# \, \mathrm{d}V
+# ```
+# The element routine computes the local  stiffness matrix `ke`
+# for a single element. `ke` is pre-allocated and reused for all elements.
+#
+# Note that the elastic stiffness tensor is constant. Thus is needs to be computed and once
+# and can then be used for all integration points.
+function assemble_cell!(ke, cellvalues, ∂σ∂ε)
+    fill!(ke, 0.0)
+
+    n_basefuncs = getnbasefunctions(cellvalues)
+    for q_point in 1:getnquadpoints(cellvalues)
+        dΩ = getdetJdV(cellvalues, q_point)
+        for i in 1:n_basefuncs
+            ∇Nᵢ = shape_gradient(cellvalues, q_point, i)# shape_symmetric_gradient(cellvalues, q_point, i)
```

Comment:
This should also be the symmetric gradient, right?

Reply:
I'm a bit undecisive what's the better way to write this. It's only valid to use symmetric gradient because of the symmetry of dsigma_depsilon. But of course that's symmetric in this case and it's also pretty common to use the symmetric version there.

Reply:
I would use the formally correct version in the implementation and elaborate in a note why the other choice is in this specific case also okay. What do you think?

Reply:
I've added a details block below with performance tips noting these possibilities.